### PR TITLE
Updated code 

### PR DESCRIPTION
**Pull request by Slowblitz**
Now the AnDO and the AnDO verbose mode works with the same function , updated test to match new function, updated version , updated documentation +typo, added new json file "isAnDO.json" with optional token, not used yet.

### CODE REVIEW

File: ando/checker.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def main():
             directory = args.path
         except IndexError:
             directory = '.'
-        error = is_AnDO_verbose(directory)
+        error = is_AnDO_verbose(directory,True)
```
Comment (SylvainTakerkart) [inline ando/checker.py]: Please add a space between the two arguments of your function: (directory, True)

File: ando/checker.py [hunk 2]
```diff
@@ -49,8 +49,8 @@ def main():
 
             directory = '.'
 
-        error_not_found = is_AnDO(directory)
-        if not error_not_found:
+        error_found = is_AnDO(directory,False)
```
Comment (SylvainTakerkart) [inline ando/checker.py]: idem

File: ando/engine.py [hunk 3]
```diff
@@ -143,81 +142,14 @@ def create_nested_list_of_path(directory):
     return nested_list_of_dir_parsed
 
 
-def is_AnDO_R(subpath, level, validate):
+def is_AnDO(directory,verbose):
```
Comment (SylvainTakerkart) [inline ando/engine.py:148]: idem

File: ando/engine.py [hunk 4]
```diff
@@ -229,11 +161,13 @@ def is_AnDO_verbose(directory):
     names = create_nested_list_of_path(directory)
 
     for item in names:
-        validate.append(is_AnDO_verbose_Format(item))
+       validate.append(is_AnDO(item,verbose))
```
Comment (SylvainTakerkart) [inline ando/engine.py:167]: idem

File: ando/tests/test_engine.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ def test_AnDO_Func_experiment_level(self):
         names = ['Landing', 'sub-anye', '180116_001_m_anye_land-001',
                  'source']
         validate = list()
-        self.assertEqual(all(andoE.is_AnDO_R(names, 0, validate)), False)
+        self.assertEqual((andoE.is_AnDO(names,False)), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_engine.py]: idem

File: ando/tests/test_engine.py [hunk 6]
```diff
@@ -45,7 +45,7 @@ def test_AnDO_Func_subject_level(self):
         names = ['exp-Landing', 'anye', 'sess-180116_001_m_anye_land-001',
                  'source']
         validate = list()
-        self.assertEqual(all(andoE.is_AnDO_R(names, 0, validate)), False)
+        self.assertEqual((andoE.is_AnDO(names,False)), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_engine.py]: idem... it should be (names, False) with a space...

File: ando/tests/test_engine.py [hunk 7]
```diff
@@ -56,7 +56,7 @@ def test_AnDO_Func_session_level(self):
         names = ['exp-Landing', 'sub-anye', '180116_001_m_anye_land-001',
                  'source']
         validate = list()
-        self.assertEqual(all(andoE.is_AnDO_R(names, 0, validate)), False)
+        self.assertEqual((andoE.is_AnDO(names,False)), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_engine.py]: idem

File: ando/tests/test_engine.py [hunk 8]
```diff
@@ -67,7 +67,7 @@ def test_AnDO_Func_sources_level(self):
         names = ['exp-Landing', 'sub-anye', 'sess-180116_001_m_anye_land-001',
                  'sources']
         validate = list()
-        self.assertEqual(all(andoE.is_AnDO_R(names, 0, validate)), False)
+        self.assertEqual((andoE.is_AnDO(names,False)), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_engine.py]: idem

File: ando/tests/test_isAnDO.py [hunk 9]
```diff
@@ -23,37 +23,37 @@ def setUp(self):
     '''testing dataset001 folder'''
     def test_AnDO_dataset_1(self):
         directory = currpath+"/dataset001/Landing"
-        self.assertEqual(andoE.is_AnDO(directory), False)
+        self.assertEqual(andoE.is_AnDO(directory,False), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_isAnDO.py]: idem

File: ando/tests/test_isAnDO.py [hunk 10]
```diff
@@ -23,37 +23,37 @@ def setUp(self):
...
 
     '''testing dataset002 folder'''
     def test_AnDO_dataset_2(self):
         directory = currpath+"/dataset002/exp-Landing"
-        self.assertEqual(andoE.is_AnDO(directory), False)
+        self.assertEqual(andoE.is_AnDO(directory,False), False)
```
Comment (SylvainTakerkart) [inline ando/tests/test_isAnDO.py]: idem!!! I stop here... can you check everywhere that you have a space between the different arguments of a function (there are several other ones below...)

File: ando/checker.py [hunk 11]
```diff
@@ -20,6 +20,24 @@
 
 dir_rules = os.path.join(os.path.dirname(__file__)) + 'rules/'
 def main():
+    """
+     
+    Main file of the validator. uses other class methods for checking
+    different aspects of the directory path.
+
+    Now uses same function for verbose and not verbose mode
```
Comment (SylvainTakerkart) [inline ando/checker.py:28]: this sentence should not be in the docstring...

File: ando/checker.py [hunk 12]
```diff
@@ -20,6 +20,24 @@
 
 dir_rules = os.path.join(os.path.dirname(__file__)) + 'rules/'
 def main():
+    """
+     
+    Main file of the validator. uses other class methods for checking
+    different aspects of the directory path.
```
Comment (SylvainTakerkart) [inline ando/checker.py:26]: Your sentence "uses other class methods for checking different aspects of the directory path." is not very clear, I don't understand it...

**Review by SylvainTakerkart — CHANGES_REQUESTED** [review]

File: ando/checker.py (hunk 12 above)
Comment (Slowblitz) [reply ando/checker.py:26]: Yes I think it's an old phrase from an old version that not relevant anymore.

File: ando/checker.py (hunk 1 above)
Comment (Slowblitz) [reply ando/checker.py]: Fixed

Comment (Slowblitz) [reply ando/checker.py]: Fixed

File: ando/checker.py (hunk 2 above)
Comment (Slowblitz) [reply ando/checker.py]: Fixed

File: ando/engine.py (hunk 3 above)
Comment (Slowblitz) [reply ando/engine.py]: Fixed

File: ando/engine.py (hunk 4 above)
Comment (Slowblitz) [reply ando/engine.py]: Fixed

File: ando/tests/test_engine.py (hunk 5 above)
Comment (Slowblitz) [reply ando/tests/test_engine.py]: Fixed

File: ando/tests/test_engine.py (hunk 6 above)
Comment (Slowblitz) [reply ando/tests/test_engine.py]: Fixed

File: ando/tests/test_engine.py (hunk 7 above)
Comment (Slowblitz) [reply ando/tests/test_engine.py]: Fixed

File: ando/tests/test_engine.py (hunk 8 above)
Comment (Slowblitz) [reply ando/tests/test_engine.py]: Fixed

File: ando/tests/test_isAnDO.py (hunk 9 above)
Comment (Slowblitz) [reply ando/tests/test_isAnDO.py]: Fixed

File: ando/checker.py (hunk 11 above)
Comment (Slowblitz) [reply ando/checker.py]: Deleted

File: ando/checker.py (hunk 12 above)
Comment (Slowblitz) [reply ando/checker.py:28]: Deleted

**Comment by SylvainTakerkart** [comment]
OK, I've accepted all your minor changes! Thanks!

Then, I cloned your branch to try it for real, but I get an error when I run the script:

AnDOChecker/ando$ python checker.py tests/dataset010/exp-Landing/
Traceback (most recent call last):
  File "checker.py", line 81, in <module>
    main()
  File "checker.py", line 69, in main
    error_found = is_AnDO(directory, False)[0]
TypeError: is_AnDO() takes 1 positional argument but 2 were given

So there seems to be a problem... Am I doing something wrong or is there a real problem?

If it's a real problem, there should be some sort of unit test that should allow detecting this automatically, no?

**Comment by Slowblitz** [comment]
It seems that I have made a mistake I have called two functions with the same name ,that leads to a malfunctioning  app I have fixed that ,unfortunately my stupidity can not be check by some unittest =)

**Comment by SylvainTakerkart** [comment]
> It seems that I have made a mistake I have called two functions with the same name ,that leads to a malfunctioning app I have fixed that ,unfortunately my stupidity can not be check by some unittest =)

OK, it seems to work for me, now ;)

However,  I don't really like the name "is_AnDO_From_Main"... you have two functions, they do different things, so their name should reflect (at least a bit) what they do... it looks like is_AnDO_From_Main is calling is_AnDO on all items of a list; so maybe the top level function (at the moment: is_AnDO_From_Main) should be called with the most simple generic name (is_AnDO); and the lower level function (at the moment: is_AnDO) should be called with a more specific name (for example: is_AnDO_item ? but maybe you can propose something better...)

**Comment by Slowblitz** [comment]
Nice =),
what about "is_AnDO"  for the top level function and "check_Path" for the lower level one ?

**Comment by SylvainTakerkart** [comment]
> Nice =),
> what about "is_AnDO" for the top level function and "check_Path" for the lower level one ?

yes, ok for me!

**Comment by Slowblitz** [comment]
Name chaged

**Pull request merged**